### DISCUSSION TITLE
Convexity of Exponential Family log-normalizer

### DISCUSSION THREAD

**Discussion by hugosou** [cvxpy/cvxpy#1775] [Ideas]
Hello.
I am trying to optimize a function containing log-normallizer of ExpFam distributions (multivariate normal in this case) which we know is convex in the natural parameters. 

Unfortunately log_det is convave and matrix_frac is convex such that I cannot build the log-normaliser without an 'UNKNOWN' curvature.

Is there a way around that that I did not think of ?
Thanks in any case.

EDIT: matrix_frac assumes the second argument is PSD. While it is NSD with natural parameters.

**Comment by SteveDiamond** [comment]
Probably you're missing a sign? Can you write out the math? This particular problem is a classic so I'm certain it can be solved if input correctly.

**Reply by hugosou** [reply]
Thanks for the quick answer. 
I am indeed pretty sure but that I am missing something...

In terms of sign:
`log_normalizer = - matrix_frac(eta1, eta2) / 4  -  log_det(-2eta2) /2`
So we're adding: convex + concave

**Reply by SteveDiamond** [reply]
Hmm maybe it's not -``log_det``? If it were ``- matrix_frac(eta1, eta2) / 4 + log_det(-2eta2) /2`` it would all be concave which seems right.

**Reply by hugosou** [reply]
It was almost a sign issue: matrix_frac assumes it second argument to be PSD but the natural parameters are negative semi-definite.
I was indeed missing something...

**Reply by rileyjmurray** [reply]
@SteveDiamond what do you think about changing ``matrix_frac`` so that it's DCP-concave when the matrix argument is NSD?

**Reply by SteveDiamond** [reply]
That's fine by me.

**Reply by hugosou** [reply]
What wasn't obvious is that I enforced asked eta2 to be NSD and `matrix_frac` did not throw an error. 
As for NSD matrices in general, a call to  `-matrix_frac(eta1, -eta2)` instead of `matrix_frac(eta1, eta2) ` worked fine.
